File: php/spec/integer.spec.php
<?php declare(strict_types=1);

use Enotype\Loaders;

describe('integer', function() {
  describe('with \'42\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::integer('42'))->toBe(42);
    });
  });
  
  describe('with \'-42\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::integer('-42'))->toBe(-42);
    });
  });
  
  describe('with \'42.0\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::integer('42.0'); })->toThrow();
    });
  });
  
  describe('with \'42,0\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::integer('42,0'); })->toThrow();
    });
  });
  
  describe('with \'4 2\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::integer('4 2'); })->toThrow();
    });
  });
  
  describe('with \'fortytwo\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::integer('fortytwo'); })->toThrow();
    });
  });
});

File: ruby/spec/integer.spec.rb
require 'enotype'

describe 'integer' do
  context 'with \'42\'' do
    it 'produces the expected result' do
      expect(Enotype::integer('42')).to be(42)
    end
  end
  
  context 'with \'-42\'' do
    it 'produces the expected result' do
      expect(Enotype::integer('-42')).to be(-42)
    end
  end
  
  context 'with \'42.0\'' do
    it 'raises an error' do
      expect { Enotype::integer('42.0') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'42,0\'' do
    it 'raises an error' do
      expect { Enotype::integer('42,0') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'4 2\'' do
    it 'raises an error' do
      expect { Enotype::integer('4 2') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'fortytwo\'' do
    it 'raises an error' do
      expect { Enotype::integer('fortytwo') }.to raise_error(RuntimeError)
    end
  end
end